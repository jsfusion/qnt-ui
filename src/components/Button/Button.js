import React from 'react';
import { PropTypes } from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { StyledButton } from './styles/Button.styles';
import { themeProps } from '../../settings';

const { classPrefix } = themeProps;

const sizes = {
  small: 'small',
  medium: 'medium',
  large: 'large'
};

const variants = {
  primary: 'primary',
  secondary: 'secondary',
};

const Button = React.forwardRef(
  (
    {
      children,
      id,
      size = sizes.medium,
      variant = 'primary',
      isDisabled = false,
      onClick,
      ...rest
    },
    ref
  ) => {
    return (
      <ThemeProvider theme={{ size: size, variant: variant }}>
        <StyledButton
          id={id}
          className={`${classPrefix}-button`}
          disabled={isDisabled}
          variant={variant}
          onClick={onClick}
        >
          <span className={`${classPrefix}-button__label`}>{children}</span>
        </StyledButton>
      </ThemeProvider>
    );
  },
);

Button.propTypes = {
  id: PropTypes.string,
  size: PropTypes.oneOf(Object.keys(sizes)),
  variant: PropTypes.oneOf(Object.keys(variants)),
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
}

export default Button;
