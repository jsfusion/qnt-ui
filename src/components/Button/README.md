# Button

Buttons are used to make common actions immediately visible and easy to perform with one click, tap, or keypress.

## Usage

```js
import { Button } from '@qnt/ui';
```

```jsx
<Button>Button label</Button>
```

## Prop Types

**id** `string`<br />
To identify the element

---
