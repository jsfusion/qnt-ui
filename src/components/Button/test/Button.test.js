import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '..';

configure({ adapter: new Adapter() });

let wrapper;
const props = {
  id: 'test_button_id',
  size: 'small',
  variant: 'secondary',
  isDisabled: false,
  onClick: () => console.log('Button clicked')
};

describe('***--- Button Component ---***', () => {
  beforeAll(() => {
    wrapper = shallow(<Button {...props} />);
  });

  it('To match snapshot of Button component', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });
});