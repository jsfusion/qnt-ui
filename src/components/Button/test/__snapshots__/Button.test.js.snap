// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`***--- Button Component ---*** To match snapshot of Button component 1`] = `ShallowWrapper {}`;
